### PR TITLE
[WNMGDS-2921] Implement Filter Chip Web Component

### PR DESCRIPTION
**Pull request by JackRyan1989**
## Summary

- [Ticket](https://jira.cms.gov/browse/WNMGDS-2921)
- Addition of `ds-filter-chip` web component
- Stories and Unit tests added

## How to test

1. Pull down this branch locally
2. Run `yarn build`
3. Run `yarn test:unit:wc` to confirm that unit tests pass
4. Run `yarn storybook` & confirm that all attributes are editable in the single filter chip example. I've disabled the props table in the multichip example because the controls didn't do anything.
5. Also in Storybook confirm that a `ds-delete` custom event is fired when clicking the chip
6. I've not added an example in Astro, but have in the CDN Web Components. You can confirm the functionality there.

## Checklist

- [X] Prefixed the PR title with the [Jira ticket number](https://jira.cms.gov/projects/WNMGDS/) as `[WNMGDS-####] Title` or [NO-TICKET] if this is unticketed work.
- [X] Selected appropriate `Type` (only one) label for this PR, if it is a breaking change, label should only be `Type: Breaking`
- [X] Selected appropriate `Impacts`, multiple can be selected.
- [X] Selected appropriate release milestone

### CODE REVIEW

**Comment by jack-ryan-nava-pbc** [comment]
Ugh. I created this while logged into my old Github account. Apparently that one still has write access?

File: packages/design-system/src/components/web-components/ds-filter-chip/ds-filter-chip.tsx
Comment (tamara-corbalt) [inline packages/design-system/src/components/web-components/ds-filter-chip/ds-filter-chip.tsx:38]: I noticed that the `label` prop is explicitly wired up here, but since it's just a string and doesn't seem to require any special handling, would it make sense to pass it through `otherProps` instead? I'm wondering if there's a specific reason for this approach.

File: packages/design-system/src/components/web-components/ds-filter-chip/ds-filter-chip.tsx
Comment (tamara-corbalt) [inline packages/design-system/src/components/web-components/ds-filter-chip/ds-filter-chip.tsx:37]: I'm curious about the semantics of passing an empty string here. Is there a specific accessibility concern we're addressing?

File: packages/design-system/src/components/web-components/ds-filter-chip/ds-filter-chip.tsx
Comment (jack-ryan-nava-pbc) [reply packages/design-system/src/components/web-components/ds-filter-chip/ds-filter-chip.tsx:38]: I think this was a typescript thing? The only required prop for FilterChip is a label, and the compiler wasn't picking up on whether it was present in the otherprops object or not, so I destructured it and explicitly passed it in to appease the typescript gods.

File: packages/design-system/src/components/web-components/ds-filter-chip/ds-filter-chip.tsx
Comment (jack-ryan-nava-pbc) [reply packages/design-system/src/components/web-components/ds-filter-chip/ds-filter-chip.tsx:37]: So the `ariaClearLabel` prop/attribute isn't _actually_ an aria attribute. It's a string that gets passed into a span element. Because the attributes for web components have to be kebab case (aria-clear-label instead of ariaClearLabel), our a11y tests correctly point out that `aria-clear-label` isn't a real aria label. So I made up a new attribute for the web component that drops the `aria-` prefix and then maps onto the `ariaClearLabel` prop when we pass it into the Preact component. Since it's an optional prop and not written onto an actual `aria` attribute anywhere, I figured it would be ok to pass an empty string to it.

**Review by tamara-corbalt — COMMENTED** [review]
Looks great so far! 
Noticing a lot of updated snapshots here. Are they the same ones as in [pr-3271](https://github.com/CMSgov/design-system/pull/3271)? If so, would it be cleaner to revert these changes and merge with main once 3271 is merged?"

File: packages/design-system/src/components/web-components/ds-filter-chip/ds-filter-chip.stories.tsx
Comment (pwolfert) [inline packages/design-system/src/components/web-components/ds-filter-chip/ds-filter-chip.stories.tsx:55]: Ask me about this in the office hours session if there's down time

Comment (jack-ryan-nava-pbc) [reply packages/design-system/src/components/web-components/ds-filter-chip/ds-filter-chip.stories.tsx:55]: Thanks for bringing this up. Closing this loop:
We are creating a ticket to do the discovery work necessary to unify close icons across the design system. This work will happen outside of this pull request.

**Review by kim-cmsds — APPROVED** [review]
The filter chip work looks good to me! I would spend a little of time verifying that the snapshot changes are all relevant to work done here to @tamara-corbalt's point.

**Review by tamara-corbalt — APPROVED** [review]
Aside from merge conflicts that need to be resolved, looks great to me!

**Pull request merged**
